### PR TITLE
[TRANSFORMATIONS] Move print_model transformation to transformation's folder

### PR DESCRIPTION
**Pull request by CuriousPanCake**
[TRANSFORMATIONS] Move print_model transformation to transformation's folder

Move print_model transformation to transformation's folder

- Tickets:
    * [CVS-151257](https://jira.devtools.intel.com/browse/CVS-151257)

### CODE REVIEW

File: src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp [hunk 1]
```diff
@@ -99,7 +99,7 @@
 #include "transformations/rt_info/keep_const_precision.hpp"
 #include "transformations/transpose_sinking/ts_shape_of.hpp"
 #include "utils/ngraph_transformation.hpp"
-#include "utils/print_model.hpp"
```
Comment (itikhono) [inline src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:102]: it's better to move it transformations/utils/print_model.hpp
all other transformations are used in some pipelines, this transformation is more for debugging and development purposes

Comment (CuriousPanCake) [reply src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:102]: Ok, moved to _transformations/utils/_

**Comment by CuriousPanCake** [comment]
No difference in .so size observed (the only .so that is rebuild with the change). The change is for free in terms of storage size.
-rwxr-xr-x 1 USER GROUP **61729496** Oct  9 11:25 libopenvino_intel_cpu_plugin.so    this PR
-rwxr-xr-x 1 USER GROUP **61729496** Oct  9 11:26 libopenvino_intel_cpu_plugin.so    master

**Comment by itikhono** [comment]
> No difference in .so size observed (the only .so that is rebuild with the change). The change is for free in terms of storage size. -rwxr-xr-x 1 USER GROUP **61729496** Oct 9 11:25 libopenvino_intel_cpu_plugin.so this PR -rwxr-xr-x 1 USER GROUP **61729496** Oct 9 11:26 libopenvino_intel_cpu_plugin.so master

Need to check the main openvino lib

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
> > No difference in .so size observed (the only .so that is rebuild with the change). The change is for free in terms of storage size. -rwxr-xr-x 1 USER GROUP **61729496** Oct 9 11:25 libopenvino_intel_cpu_plugin.so this PR -rwxr-xr-x 1 USER GROUP **61729496** Oct 9 11:26 libopenvino_intel_cpu_plugin.so master
> 
> Need to check the main openvino lib

Code is header only and not included to OpenVINO sources => no binary size increase.

**Review by itikhono — APPROVED** [review]

**Pull request merged**